### ISSUE TITLE
Keyboard focus is not moved correctly after opening a subpage with keyboard navigation

### ISSUE THREAD

**Issue by kalairubachandrakumaran-oss** [digitalfabrik/integreat-app#4213]
### Describe the Bug
Keyboard navigation generally works but following issues exists:
- If I open a tile using touch, I can navigate through the individual menu items on the next page with the arrow keys
- If I open the same tile using the arrow keys (keyboard), I can open the page, but I cannot navigate through the individual menu items on that page using the arrow keys

### Steps to Reproduce

1. Connect an external keyboard to an iOS device
2. Open the integreat app
3. Navigate to a location using keyboard
4. Navigate to a tile using the arrow keys and select a tile with keyboard

### Expected Behavior
The Keyboard focus should move to the new and all menu items should be navigable using arrow keys

### Environment

- OS version:
    - iOS version 26.5 (iPhone 12 mini)
- App version: 2026.6.0

### Related Issues
#4084 

### Additional Information

https://github.com/user-attachments/assets/9dca0a5a-c231-4ce0-bb97-fedbd06ebad3

**Comment by steffenkleinle** [comment]
@kalairubachandrakumaran-oss this is iOS only, right?

**Comment by kalairubachandrakumaran-oss** [comment]
> [@kalairubachandrakumaran-oss](https://github.com/kalairubachandrakumaran-oss) this is iOS only, right?
> yes

**Comment by bahaaTuffaha** [comment]
Couldn't find a fix other than one and it will create a bit of a mess:
- at `useHeader` currently we pass the header to navigation.setOptions => header . so the fix here to return the header and include it above other elements like a normal component.
~~I will push a draft with these changes.~~

**Comment by bahaaTuffaha** [comment]
Let's wait and see if future updates could fix the problem.

**Comment by steffenkleinle** [comment]
@bahaaTuffaha any chance this got fixed by #4300?

**Comment by bahaaTuffaha** [comment]
> [@bahaaTuffaha](https://github.com/bahaaTuffaha) any chance this got fixed by [#4300](https://github.com/digitalfabrik/integreat-app/pull/4300)?

For some reason on iOS 26.1 when I enable the full keyboard mode it doesn't highlight anything not even the system ui. On 26.5 the system navigation with keyboard worked but on the app nothing is highlighted. 😵‍💫
